lib/interactive_viewer.ml

```diff
@@ -60,9 +60,41 @@ let navigate z_patches (dir : direction) : unit =
   | Prev -> Lwd.set z_patches (Zipper.prev z)
   | Next -> Lwd.set z_patches (Zipper.next z)
 
-(* let pure_str s = Lwd.pure (W.string s) *)
 let quit = Lwd.var false
 
+let additions_and_removals lines =
+  let add_line (additions, removals) line =
+    match line with
+    | `Their _ -> (additions + 1, removals)
+    | `Mine _ -> (additions, removals + 1)
+    | `Common _ -> (additions, removals)
+  in
+  List.fold_left add_line (0, 0) lines
+
+let accumulate_count hunks =
+  List.fold_left
+    (fun (add_acc, remove_acc) hunk ->
+      let add_in_hunk, remove_in_hunk =
+        additions_and_removals hunk.Patch.lines
+      in
+      (add_acc + add_in_hunk, remove_acc + remove_in_hunk))
+    (0, 0) hunks
+
+let change_summary z_patches : ui Lwd.t =
+  let$ z = Lwd.get z_patches in
+  let p = Zipper.get_focus z in
+  let total_additions, total_removals = accumulate_count p.Patch.hunks in
+  let format_plural n singular plural =
+    if n = 1 then Printf.sprintf "%d %s" n singular
+    else Printf.sprintf "%d %s" n plural
+  in
+  let operation_count =
+    Printf.sprintf "%s, %s"
+      (format_plural total_additions "addition" "additions")
+      (format_plural total_removals "removal" "removals")
+  in
+  W.string ~attr:Notty.A.(fg lightcyan) @@ Printf.sprintf "%s\n" operation_count
```

Comment:
This `\n` is used to render an empty line. Is it intended ?

If it is, I'd prefer it not to be implemented like this. Instead an empty line should be explicitly added in the `W.vbox` with `W.string "";`.

In any case, this should be simplified to:
```suggestion
  W.string ~attr:Notty.A.(fg lightcyan) operation_count
```

Reply:
@Julow, Thank you for your feedback! I'm glad to hear that you find the implementation satisfactory.
I'll address the comment regarding the empty line promptly. Once that's done, I'll let you know.

Reply:
I agree the output looks better without the newline.

I do not see any new modification in the PR, did you push to the same branch ?

Reply:
@Julow wanted to check before making the change, the change should be in and ready to merge.